apps/condo/domains/user/schema/SignInUserService.js

```diff
@@ -0,0 +1,221 @@
+ * @param {string[]} requiredFields
+ * @param {*} userData
+ * @return {{missingFields: string[], missingRequiredFields: boolean}}
+ */
+const checkRequiredUserFields = (requiredFields, userData) => {
+    if (!Array.isArray(requiredFields)) throw new Error('"requiredFields" should be an array')
+
+    const missingFields = []
+    for (const field of requiredFields) {
+        const value = get(userData, field, null)
+        if (isNil(value)) missingFields.push(field)
+        if (typeof value === 'string' && value.length < 1) missingFields.push(field)
+    }
+    return { missingRequiredFields: missingFields.length > 0, missingFields }
+}
+
+const SignInUserService = new GQLCustomSchema('SignInUserService', {
+    types: [
+        {
+            access: true,
+            type: 'input SignInUserUserDataInput { phone: String, name: String, email: String, password: String, meta: JSON }',
+        },
+        {
+            access: true,
+            type: 'input SignInUserInput { dv: Int!, sender: SenderFieldInput!, confirmActionToken: ID!, userType: UserTypeType!, userData: SignInUserUserDataInput }',
+        },
+        {
+            access: true,
+            type: 'type SignInUserOutput { user: User, token: ID! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canSignInUser,
+            schema: 'signInUser(data: SignInUserInput!): SignInUserOutput',
+            doc: {
+                summary: 'This mutation authorizes the user by token, after confirming the phone number.\n\n' +
+                    'If the user is not registered, then he will be created with the data that is passed in the payload (user data).\n\n' +
+                    'If the existing user is missing some fields, then these fields will be taken from the payload (user data) and updated.\n\n' +
+                    'This mutation is not available for service users!',
+                errors: Object.fromEntries(Object.entries(ERRORS).filter(([, error]) => typeof error === 'object')),
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { confirmActionToken, userType, userData, dv, sender } = data
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                if (!SUPPORTED_USER_TYPES.includes(userType)) throw new GQLError(ERRORS.NOT_SUPPORTED_USER_TYPE, context)
+
+                if (!confirmActionToken) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                await checkDailyRequestLimitCountersByIp(context, 'signInUser', context.req.ip)
+
+                const action = await ConfirmPhoneAction.getOne(context,
+                    {
+                        token: confirmActionToken,
+                        expiresAt_gte: new Date().toISOString(),
+                        completedAt: null,
+                        isPhoneVerified: true,
```

Comment:
what about isEmail* fields?

Reply:
it should depends on Token type?

Reply:
Or we will create another token?

Reply:
We don't support email yet, so there is no logic with email here. After the email token adding, the logic will be updated